spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/nn/MergeSpec.scala

```diff
@@ -0,0 +1,95 @@
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.bigdl.utils.T
+
+class MergeSpec extends KerasBaseSpec {
+
+  "Merge sum" should "work properly" in {
+    val input1 = Tensor[Float](2, 4, 8).rand(0, 1)
+    val input2 = Tensor[Float](2, 4, 8).rand(0, 1)
+    val input = T(1 -> input1, 2 -> input2)
+    val seq = KSequential[Float]()
+    val l1 = InputLayer[Float](inputShape = Shape(4, 8))
+    val l2 = InputLayer[Float](inputShape = Shape(4, 8))
+    val layer = Merge[Float](layers = List(l1, l2), "sum")
+    seq.add(layer)
+    seq.getOutputShape().toSingle().toArray should be (Array(-1, 4, 8))
+    seq.forward(input) should be (input1 + input2)
+  }
+
+  "Merge ave" should "work properly" in {
+    val input1 = Tensor[Float](3, 10).rand(0, 1)
+    val input2 = Tensor[Float](3, 10).rand(0, 1)
+    val input3 = Tensor[Float](3, 10).rand(0, 1)
+    val input = T(1 -> input1, 2 -> input2, 3 -> input3)
+    val seq = KSequential[Float]()
+    val l1 = InputLayer[Float](inputShape = Shape(10))
+    val l2 = InputLayer[Float](inputShape = Shape(10))
+    val l3 = InputLayer[Float](inputShape = Shape(10))
+    val layer = Merge[Float](layers = List(l1, l2, l3), "ave")
+    seq.add(layer)
+    seq.getOutputShape().toSingle().toArray should be (Array(-1, 10))
+    seq.forward(input) should be ((input1 + input2 + input3)/3)
+  }
+
+  "Merge concat" should "work properly" in {
+    val input1 = Tensor[Float](2, 3, 8).rand(0, 1)
+    val input2 = Tensor[Float](2, 4, 8).rand(0, 1)
+    val input = T(1 -> input1, 2 -> input2)
+    val seq = KSequential[Float]()
+    val l1 = InputLayer[Float](inputShape = Shape(3, 8))
+    val l2 = InputLayer[Float](inputShape = Shape(4, 8))
+    val layer = Merge[Float](layers = List(l1, l2), "concat", concatAxis = 1)
+    seq.add(layer)
+    seq.getOutputShape().toSingle().toArray should be (Array(-1, 7, 8))
+    seq.forward(input)
+  }
+
+  "Merge dot" should "work properly" in {
+    val input1 = Tensor[Float](2, 4).rand(0, 1)
+    val input2 = Tensor[Float](2, 4).rand(0, 1)
+    val input = T(1 -> input1, 2 -> input2)
+    val seq = KSequential[Float]()
+    val l1 = InputLayer[Float](inputShape = Shape(4))
+    val l2 = InputLayer[Float](inputShape = Shape(4))
+    val layer = Merge[Float](layers = List(l1, l2), "dot")
+    seq.add(layer)
+    seq.getOutputShape().toSingle().toArray should be (Array(-1, 1))
+    seq.forward(input)
+  }
+
+  "Merge complicated" should "work properly" in {
+    val input1 = Tensor[Float](3, 8).rand(0, 1)
+    val input2 = Tensor[Float](3, 6).rand(0, 1)
+    val input = T(1 -> input1, 2 -> input2)
+    val seq = KSequential[Float]()
+    val l1 = Dense[Float](10, inputShape = Shape(8))
+    val l2 = Dense[Float](10, inputShape = Shape(6))
+    val layer = Merge[Float](layers = List(l1, l2), "max")
+    seq.add(layer)
+    seq.add(Dense[Float](15))
+    seq.getOutputShape().toSingle().toArray should be (Array(-1, 15))
+    seq.forward(input)
```

Comment:
validation?

Reply:
Can't be validated on scala side. Will compare with Keras on python side.